notebooks/generative-ai/lib/elasticsearch_chat_message_history.py

```diff
@@ -0,0 +1,81 @@
+import json
+import logging
+from typing import List
+from elasticsearch import ApiError, Elasticsearch
+
+from langchain.schema import BaseChatMessageHistory
+from langchain.schema.messages import BaseMessage, _message_to_dict, messages_from_dict
+
+logger = logging.getLogger(__name__)
+
+class ElasticsearchChatMessageHistory(BaseChatMessageHistory):
+    """Chat message history that stores history in Elasticsearch.
+
+    Args:
+        client: Elasticsearch client.
+        index: name of the index to use.
+        session_id: arbitrary key that is used to store the messages
+            of a single chat session.
+    """
+
+    def __init__(
+        self,
+        client: Elasticsearch,
```

Comment:
one thing to keep in mind - in langchain they dynamically import the Elasticsearch client when needed as its optionally imported. See example here  https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/vectorstores/elastic_vector_search.py#L139-L145

Reply:
Ok, switched to dynamic import.